### PR TITLE
BASH-16 Add to eslint config

### PR DESCRIPTION
**Pull request by csduarte**
Add more rules to eslint config and packaging rules.

- [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,17 +1,169 @@
 {
-    "extends": "./.eslintrc-platform.json",
```
Comment (yuya-oc) [inline .eslintrc.json]: The purpose is to use same base config with mattermost-webapp. You can write only additional rules here. What was actually needed for you?

File: package.json [hunk 2]
```diff
@@ -31,7 +31,8 @@
     "package:mac": "build --mac --publish=never",
     "package:linux": "build --linux --x64 --ia32 --em.name=mattermost-desktop --publish=never",
     "manipulate-windows-zip": "node scripts/manipulate_windows_zip.js",
-    "lint:js": "eslint --ext .js --ext .jsx ."
+    "lint:js": "eslint --ext .js --ext .jsx .",
+    "eslint": "eslint --ext .js,.jsx src/browser src/common src/main"
```
Comment (yuya-oc) [inline package.json]: Why is this script added?

**Comment by csduarte** [comment]
@dmeza I couldn't say for sure, I'm not the one who added it. But if there is a platform specific eslint, will the editor pick up on it?

**Comment by jasonblais** [comment]
> But if there is a platform specific eslint, will the editor pick up on it?

@yuya-oc thoughts on this question?

**Comment by yuya-oc** [comment]
Honestly I'm not sure what I should say now though, `"extends"` is a feature of eslint. https://eslint.org/docs/user-guide/configuring#extending-configuration-files So any editors should use both of `.eslintrc-platform.json`(base config) and `.eslintrc.json`(override).

In `.eslintrc.json`, we can write only desktop app specific rules. However this PR would make difficult to know the rules.

**Comment by csduarte** [comment]
Okay closing, will reopen if we discover any editor issues (which is why I think we created in the first place)

**Pull request closed without merging**